File: packages/helix-persistence/src/helix_agent/persistence/models/platform_embedding_config.py
"""Platform embedding/rerank config ORM model — Stream T (PR B).

A single-row (``id == "singleton"``) table storing the platform's chosen
embedding / rerank provider+model selection. This is **non-secret** config
(provider/model names only — actual API keys live in ``platform_provider_secret``).

Platform-global, tenant-less (like ``platform_provider_secret`` /
``role_binding`` platform-scope rows) — no RLS policy is attached; all access
goes through ``bypass_rls_session()``. An absent row means "not configured".
"""

from __future__ import annotations

from datetime import datetime

from sqlalchemy import DateTime, Text, func
from sqlalchemy.orm import Mapped, mapped_column

from helix_agent.persistence.base import Base


class PlatformEmbeddingConfigRow(Base):
    """The single platform embedding/rerank selection row."""

    __tablename__ = "platform_embedding_config"

    id: Mapped[str] = mapped_column(Text, primary_key=True)
    embedding_provider: Mapped[str | None] = mapped_column(Text, nullable=True)
    embedding_model: Mapped[str | None] = mapped_column(Text, nullable=True)
    rerank_provider: Mapped[str | None] = mapped_column(Text, nullable=True)
    rerank_model: Mapped[str | None] = mapped_column(Text, nullable=True)
    updated_at: Mapped[datetime] = mapped_column(
        DateTime(timezone=True), nullable=False, server_default=func.now()
    )
    updated_by: Mapped[str | None] = mapped_column(Text, nullable=True)

File: apps/admin-ui/src/api/platform_embedding_config.ts
/**
 * Platform Embedding/Rerank config SDK — backed by /v1/platform/embedding-config
 * (Stream T PR C). system_admin-only, platform-level.
 */
import { getJson, putJson } from "./client";

export interface ProviderModel {
  provider: string;
  model: string;
}

export interface PlatformEmbeddingConfigView {
  embedding: ProviderModel | null;
  rerank: ProviderModel | null;
  available_embedding: ProviderModel[];
  available_rerank: ProviderModel[];
}

export interface PlatformEmbeddingConfigWrite {
  embedding_provider: string;
  embedding_model: string;
  rerank_provider?: string;
  rerank_model?: string;
}

export async function getPlatformEmbeddingConfig(): Promise<PlatformEmbeddingConfigView> {
  return getJson<PlatformEmbeddingConfigView>("/v1/platform/embedding-config");
}

export async function putPlatformEmbeddingConfig(
  body: PlatformEmbeddingConfigWrite,
): Promise<{ embedding: ProviderModel | null; rerank: ProviderModel | null }> {
  return putJson("/v1/platform/embedding-config", body);
}
